src/main/java/net/mcreator/ui/modgui/ModElementGUI.java

```diff
@@ -525,12 +525,13 @@ private void showErrorsMessage(AggregatedValidationResult validationResult) {
 	/**
 	 * This method implements the mod element saving and generation
 	 */
-	private void finishModCreation(boolean closeTab) {
+	@SuppressWarnings("EqualsBetweenInconvertibleTypes") private void finishModCreation(boolean closeTab) {
 		MCREvent.event(new ModElementGUIEvent.WhenSaving(mcreator, tabIn, this, !closeTab));
 		GE element = getElementFromGUI();
 
 		// if new element, and if we are not in the root folder, specify the folder of the mod element
-		if (!editingMode && !mcreator.mv.currentFolder.equals(mcreator.getWorkspace().getFoldersRoot()))
+		if (!editingMode && mcreator.getWorkspace().getFoldersRoot().getRecursiveFolderChildren()
```

Comment:
I don't understand the need for this change. The purpose of this if was to not set folder it the folder is "root", now you changed this behavior here

Reply:
Ok, but root check is still needed.

It would also help to document those changes in the PR description

Reply:
On it.

Reply:
This could also be a method in `WorkspacePanel` (just thinking aloud, doing it in this PR would be too much).

Reply:
ModElement could also check if it is being set to root path to avoid having to worry about this check (for another PR)

Reply:
Ok nevermind this is already the case